cmd/machine-config-daemon/start.go

```diff
@@ -167,6 +169,25 @@ func runStartCmd(cmd *cobra.Command, args []string) {
 		return
 	}
 
+	// Use kubelet kubeconfig file to get the URL to kube-api-server
+	kubeconfig, err := clientcmd.LoadFromFile("/etc/kubernetes/kubeconfig")
+	if err != nil {
+		glog.Errorf("failed to load kubelet kubeconfig: %v", err)
+	}
+	clusterName := kubeconfig.Contexts[kubeconfig.CurrentContext].Cluster
+	apiURL := kubeconfig.Clusters[clusterName].Server
+
+	url, err := url.Parse(apiURL)
+	if err != nil {
+		glog.Errorf("failed to parse api url from kubelet kubeconfig: %v", err)
```

Comment:
This should be `Fatalf` I believe; no point in just stumbling forward and fail to parse an empty string URL.

(In Rust, this wouldn't be possible because the `Result` type is an enum, so you can't have the case of an error being returned but still trying to use the success value...)